internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -2651,3 +2704,66 @@ func flattenKubernetesClusterMaintenanceConfigurationTimeInWeeks(input *[]contai
 	}
 	return results
 }
+
+func expandKubernetesClusterHttpProxyConfig(input []interface{}) *containerservice.ManagedClusterHTTPProxyConfig {
+	httpProxyConfig := containerservice.ManagedClusterHTTPProxyConfig{}
+	if len(input) == 0 {
+		return nil
+	}
+
+	config := input[0].(map[string]interface{})
+	httpProxyConfig.HTTPProxy = utils.String(config["http_proxy"].(string))
+	httpProxyConfig.HTTPSProxy = utils.String(config["https_proxy"].(string))
+	httpProxyConfig.TrustedCa = utils.String(config["trusted_ca"].(string))
+	noProxyRaw := config["no_proxy"].([]interface{})
+	httpProxyConfig.NoProxy = utils.ExpandStringSlice(noProxyRaw)
+	return &httpProxyConfig
+}
+
+func flattenKubernetesClusterHttpProxyConfig(props *containerservice.ManagedClusterProperties, flattenedDefaultNodePool *[]interface{}) []interface{} {
+	results := make(map[string]interface{})
+	httpProxyConfig := props.HTTPProxyConfig
+	if httpProxyConfig != nil {
+		results["http_proxy"] = httpProxyConfig.HTTPProxy
+		results["https_proxy"] = httpProxyConfig.HTTPSProxy
+		noProxyList := httpProxyConfig.NoProxy
+		// Removing the default values that AKS always applies from the flatten
+		newNoProxyList := make([]string, len(*noProxyList))
+		ignoreList := []string{
+			"168.63.129.16",
+			"169.254.169.254",
+			"127.0.0.1",
+			"localhost",
+			"konnectivity",
+		}
+		if props.NetworkProfile.PodCidr != nil && *props.NetworkProfile.PodCidr != "" {
+			ignoreList = append(ignoreList, *props.NetworkProfile.PodCidr)
+		}
+		if props.NetworkProfile.ServiceCidr != nil && *props.NetworkProfile.ServiceCidr != "" {
+			ignoreList = append(ignoreList, *props.NetworkProfile.ServiceCidr)
+		}
+		if props.NetworkProfile.DockerBridgeCidr != nil && *props.NetworkProfile.DockerBridgeCidr != "" {
+			ignoreList = append(ignoreList, *props.NetworkProfile.DockerBridgeCidr)
+		}
+		if props.Fqdn != nil && *props.Fqdn != "" {
+			ignoreList = append(ignoreList, *props.Fqdn)
+		}
+		if (*flattenedDefaultNodePool)[0] != nil && (*flattenedDefaultNodePool)[0].(map[string]interface{})["vnet_subnet_id"] == "" {
+			ignoreList = append(ignoreList, "10.240.0.0/12")
+		}
+
+		i := 0
+		for _, element := range *noProxyList {
+			if !utils.SliceContainsValue(ignoreList, element) {
+				newNoProxyList[i] = element
+				i++
+			}
+		}
+		newNoProxyList = newNoProxyList[:i]
+		results["no_proxy"] = newNoProxyList
+		results["trusted_ca"] = httpProxyConfig.TrustedCa
```

Comment:
ditto here, this is a pointer so could be nil in the API response however we require that a value is always set:

```suggestion
		trustedCertificateAuthority := ""
		if httpProxyConfig.TrustedCa != nil {
			trustedCertificateAuthority = *httpProxyConfig.TrustedCa
		}
		results["trusted_ca"] = trustedCertificateAuthority
```